openshift-kube-apiserver/admission/autoscaling/managementcpusoverride/admission.go

```diff
@@ -0,0 +1,323 @@
+package managementcpusoverride
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apiserver/pkg/admission"
+	"k8s.io/apiserver/pkg/admission/initializer"
+	"k8s.io/client-go/informers"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/klog/v2"
+	coreapi "k8s.io/kubernetes/pkg/apis/core"
+	"k8s.io/kubernetes/pkg/kubelet/cm"
+)
+
+const (
+	PluginName                              = "autoscaling.openshift.io/ManagementCPUsOverride"
+	podWorkloadAnnotation                   = "workload.openshift.io"
+	podWorkloadAnnotationEffect             = "effect"
+	workloadEffectPreferredDuringScheduling = "PreferredDuringScheduling"
+	managedCapacityLabel                    = "management.workload.openshift.io/cores"
+	containerResourcesAnnotationPrefix      = "io.openshift.workload"
+	containerCPUSharesAnnotation            = "io.openshift.workload.management.cpushares"
+	containerCPUSetAnnotation               = "io.openshift.workload.management.cpuset"
+	namespaceAllowedAnnotation              = "workload.openshift.io/allowed"
```

Comment:
We had a discussion to mark all labels internal to avoid turning this into an API we have to support forever. I don't see "internal" anymore in any of the annotations or labels. Where has that discussion gone?

Reply:
As far as I know, the final decision was to use annotations without the prefix. @smarterclayton @browsell Can you please confirm?

Reply:
Correct.  This feature can be used by user workloads going forward.